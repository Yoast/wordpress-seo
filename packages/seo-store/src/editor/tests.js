import editorReducer, { editorActions } from "./slice";

describe( "Editor slice", () => {
	// eslint-disable-next-line no-undefined
	const previousState = undefined;

	const initialState = {
		content: "",
		title: "",
		permalink: "",
		excerpt: "",
		date: "",
		featuredImage: {},
		locale: "",
	};

	describe( "Reducer", () => {
		test( "should return the editor slice its initial state", () => {
			expect( editorReducer( previousState, {} ) ).toEqual( initialState );
		} );

		test( "should update the content", () => {
			const { updateContent } = editorActions;

			const result = editorReducer( previousState, updateContent( "test" ) );

			expect( result ).toEqual( {
				...initialState,
				content: "test",
			} );
		} );

		test( "should update the title", () => {
			const { updateTitle } = editorActions;

			const result = editorReducer( initialState, updateTitle( "test" ) );

			expect( result ).toEqual( {
				...initialState,
				title: "test",
			} );
		} );

		test( "should update the permalink", () => {
			const { updatePermalink } = editorActions;

			const result = editorReducer( initialState, updatePermalink( "test" ) );

			expect( result ).toEqual( {
				...initialState,
				permalink: "test",
			} );
		} );

		test( "should update the excerpt", () => {
			const { updateExcerpt } = editorActions;

			const result = editorReducer( initialState, updateExcerpt( "test" ) );

			expect( result ).toEqual( {
				...initialState,
				excerpt: "test",
			} );
		} );

		test( "should update the date", () => {
			const { updateDate } = editorActions;

			const result = editorReducer( initialState, updateDate( new Date( Date.UTC( 2021, 11, 15, 12 ) ).toISOString() ) );

			expect( result ).toEqual( {
				...initialState,
				date: "2021-12-15T12:00:00.000Z",
			} );
		} );

		test( "should update the featured image", () => {
			const { updateFeaturedImage } = editorActions;

			const result = editorReducer( initialState, updateFeaturedImage( { url: "http://example.com" } ) );

			expect( result ).toEqual( {
				...initialState,
				featuredImage: {
					url: "http://example.com",
				},
			} );
		} );

		test( "should update the locale", () => {
			const { updateLocale } = editorActions;

			const result = editorReducer( initialState, updateLocale( "ja" ) );

			expect( result ).toEqual( {
				...initialState,
				locale: "ja",
			} );
		} );
	} );
} );
