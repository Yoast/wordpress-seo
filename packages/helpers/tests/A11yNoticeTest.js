import React from "react";
import renderer from "react-test-renderer";

import { A11yNotice } from "../src/tests/A11yNotice";

test( "the A11yNotice matches the snapshot", () => {
	const component = renderer.create(
		<A11yNotice>A11yNoticeValue</A11yNotice>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );


test( "the A11yNotice can be changed to a different element", () => {
	const H2A11yNotice = A11yNotice.withComponent( "h2" );

	const component = renderer.create(
		<H2A11yNotice>H2A11yNoticeValue</H2A11yNotice>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
	expect( tree.type ).toEqual( "h2" );
} );
