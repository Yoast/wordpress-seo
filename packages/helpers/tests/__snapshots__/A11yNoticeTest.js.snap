// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the A11yNotice can be changed to a different element 1`] = `
.c0 {
  border: 0;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute !important;
  width: 1px;
  word-wrap: normal !important;
  -webkit-transform: translateY(1em);
  -ms-transform: translateY(1em);
  transform: translateY(1em);
}

<h2
  className="c0"
>
  H2A11yNoticeValue
</h2>
`;

exports[`the A11yNotice matches the snapshot 1`] = `
.c0 {
  border: 0;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute !important;
  width: 1px;
  word-wrap: normal !important;
  -webkit-transform: translateY(1em);
  -ms-transform: translateY(1em);
  transform: translateY(1em);
}

<span
  className="c0"
>
  A11yNoticeValue
</span>
`;
